sheepdog/auth/__init__.py

```diff
@@ -31,15 +33,28 @@
 
 
 def get_jwt_from_header():
-    jwt = None
+    jwt_token = None
     auth_header = flask.request.headers.get("Authorization")
     if auth_header:
         items = auth_header.split(" ")
         if len(items) == 2 and items[0].lower() == "bearer":
-            jwt = items[1]
-    if not jwt:
+            jwt_token = items[1]
+    if not jwt_token:
+        raise AuthNError("Didn't receive JWT correctly")
+    return jwt_token
+
+
+def check_if_jwt_expired(jwt_token):
+    """
+    Check if the JWT is expired.
```

Comment:
i think the point isn't exactly to check if the token is expired right now, but to check if it will expire before the cache does.
So, suggesting something like this:
```
CACHE_SECS = 1
AUTHZ_CACHE = SimpleCache(default_timeout=CACHE_SECS)
[...]
def check_if_jwt_close_to_expiration(jwt_token): # renamed function
    [...]
    return (decoded_token.get("exp", 0) + CACHE_SECS) < time.time()
```

Reply:
Thanks for bringing this up. The idea makes more sense to me now, however I think we should check if the token is going to expire before the cache expires. So more appropriate logic must be 
```
 return (decoded_token.get("exp", 0)  < time.time() + CACHE_SECS)
 ```
 
 Correct me if I'm wrong.